File: tests/fixtures/while_loop.clj
; Test: While loop with array initialization

class Item {
    def value:int 0
    
    Constructor (v:int) {
        value = v
    }
    
    sfn create:Item (v:int) {
        return (new Item(v))
    }
}

class Container {
    def items:[Item]
    
    Constructor () {
        this.init()
    }
    
    fn init:void () {
        def i 0
        while (i < 5) {
            push items (Item.create(i))
            i = i + 1
        }
    }
    
    fn getCount:int () {
        return (array_length items)
    }
}

class WhileLoopTest {
    sfn m@(main):void () {
        def c (new Container())
        def cnt (c.getCount())
        print ("Count: " + cnt)
        print "Done"
    }
}

File: tests/output-rust/while_loop.rs
#![allow(unused_parens)]
#![allow(unused_mut)]
#![allow(unused_variables)]
#![allow(non_snake_case)]
#![allow(dead_code)]

#[derive(Clone)]
struct Item { 
  value : i64, 
}
impl Item { 
  
  pub fn new(v : i64) ->  Item {
    let mut me = Item { 
      value:0, 
    };
    me.value = v;
    return me;
  }
  pub fn create(v : i64) -> Item {
    return Item::new(v).clone();
  }
}
#[derive(Clone)]
struct Container { 
  items : Vec<Item>, 
}
impl Container { 
  
  pub fn new() ->  Container {
    let mut me = Container { 
      items: Vec::new(), 
    };
    me.init();
    return me;
  }
  fn init(&mut self, ) -> () {
    let mut i : i64 = 0;
    while i < 5 {
      self.items.push(Item::create(i));
      i = i + 1;
    }
  }
  fn getCount(&mut self, ) -> i64 {
    return self.items.len() as i64;
  }
}
#[derive(Clone)]
struct WhileLoopTest { 
}
impl WhileLoopTest { 
  
  pub fn new() ->  WhileLoopTest {
    let mut me = WhileLoopTest { 
    };
    return me;
  }
}
fn main() {
  let mut c : Container = Container::new();
  let cnt : i64 = c.getCount();
  println!( "{}", ["Count: ".to_string() , (cnt.to_string()) ].join("") );
  println!( "{}", "Done".to_string() );
}
